js/zb.js

```diff
@@ -969,21 +969,117 @@ module.exports = class zb extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
-        if (type !== 'limit') {
-            throw new InvalidOrder (this.id + ' allows limit orders only');
-        }
         await this.loadMarkets ();
+        const market = this.market (symbol);
+        const swap = market['swap'];
+        const spot = market['spot'];
+        const action = params['action'];
+        const timeInForce = this.safeString (params, 'timeInForce');
+        if (spot) {
+            if (type !== 'limit') {
+                throw new InvalidOrder (this.id + ' allows limit orders only');
+            }
+        }
+        if (type === 'market') {
+            throw new InvalidOrder (this.id + ' does not allow market orders');
+        }
+        const method = this.getSupportedMapping (market['type'], {
+            'spot': 'spotV1PrivateGetOrder',
+            'swap': 'contractV2PrivatePostTradeOrder',
+        });
         const request = {
-            'price': this.priceToPrecision (symbol, price),
             'amount': this.amountToPrecision (symbol, amount),
-            'tradeType': (side === 'buy') ? '1' : '0',
-            'currency': this.marketId (symbol),
-        };
-        const response = await this.spotV1PrivateGetOrder (this.extend (request, params));
-        return {
-            'info': response,
-            'id': response['id'],
         };
+        if (price) {
+            request['price'] = this.priceToPrecision (symbol, price);
+        }
+        if (spot) {
+            request['tradeType'] = (side === 'buy') ? '1' : '0';
+            request['currency'] = market['id'];
+        } else if (swap) {
+            const reduceOnly = this.safeValue (params, 'reduceOnly');
+            if (side === 'sell' && reduceOnly || side === 3) {
+                request['side'] = 3;
+            } else if (side === 'sell' && reduceOnly || side === 4) {
+                request['side'] = 4;
+            } else if (side === 'buy' || side === 1) {
+                request['side'] = 1;
+            } else if (side === 'buy' || side === 2) {
+                request['side'] = 2;
+            }
```

Comment:
The above `side` conditions don't look right to me. If `side` is 'sell' and `reduceOnly` is set, then we never execute line 1004. Also, if side is `buy`, we never execute line 1008, which is strange.
I think something must be rewritten there. In fact, the comments i've outlined above for `params['action']` also apply here to the `params['side']` as well. We can simplify this if-else-spaghetti significantly.

Reply:
I'm not sure the best way to handle this, there's open long, open short, close long, and close short types. I think instead of reduceOnly which would specify if it's closing the position, I need a way to specify whether the order is for a long or short position. Maybe I can check whether the price or amount argument is negative to determine if it's long or short then convert it back to positive to make it work with the API. Let me know if you have a better idea of how to handle this

Reply:
I adjusted this to require the user to set the params['side'] to the corresponding integer as well as the side argument, Let me know if you think there is a better approach